### PR TITLE
add sqlglot benchmarks

### PR DESCRIPTION
**Pull request by tobymao**
[sqlglot](https://github.com/tobymao/sqlglot) is a pure python sql parser, transpiler, and optimizer

### CODE REVIEW

**Comment by mdboom** [comment]
> @mdboom do you need anything from me in order to land these changes?

This looks fine to me, but I don't have merge rights. Maybe @ericsnowcurrently can have a look.

File: pyperformance/data-files/benchmarks/bm_sqlglot/run_benchmark.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    return pyperf.perf_counter() - timer
+
+
+if __name__ == "__main__":
+    runner = pyperf.Runner()
+    runner.metadata['description'] = "SQLGlot benchmark"
+    runner.bench_time_func("parse", bench_sqlglot, bench_parse)
+    runner.bench_time_func("transpile", bench_sqlglot, bench_transpile)
+    runner.bench_time_func("optimize", bench_sqlglot, bench_optimize)
+    runner.bench_time_func("normalize", bench_sqlglot, bench_normalize)
```
Comment (ericsnowcurrently) [inline pyperformance/data-files/benchmarks/bm_sqlglot/run_benchmark.py:162]: Make sure to use an appropriate prefix on the names:
```suggestion
    runner.bench_time_func("sqlglot_parse", bench_sqlglot, bench_parse)
    runner.bench_time_func("sqlglot_transpile", bench_sqlglot, bench_transpile)
    runner.bench_time_func("sqlglot_optimize", bench_sqlglot, bench_optimize)
    runner.bench_time_func("sqlglot_normalize", bench_sqlglot, bench_normalize)
```

File: pyperformance/data-files/benchmarks/bm_sqlglot/run_benchmark.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    normalize.normalize(conjunction)
+
+
+def bench_sqlglot(loops, func):
+    timer = pyperf.perf_counter()
+
+    for _ in range(loops):
+        func()
+
+    return pyperf.perf_counter() - timer
```
Comment (ericsnowcurrently) [inline pyperformance/data-files/benchmarks/bm_sqlglot/run_benchmark.py:153]: Consider benchmarking a bit tighter:

```suggestion
def bench_parse(loops):
    elapsed = 0
    for _ in range(loops):
        t0 = pyperf.perf_counter()
        parse_one(SQL)
        elapsed += pyperf.perf_counter() - t0
    return elapsed


def bench_transpile(loops):
    elapsed = 0
    for _ in range(loops):
        t0 = pyperf.perf_counter()
        transpile(SQL, write="spark")
        elapsed += pyperf.perf_counter() - t0
    return elapsed


def bench_optimize():
    elapsed = 0
    for _ in range(loops):
        t0 = pyperf.perf_counter()
        optimize(parse_one(SQL), TPCH_SCHEMA)
        elapsed += pyperf.perf_counter() - t0
    return elapsed


def bench_normalize():
    elapsed = 0
    for _ in range(loops):
        t0 = pyperf.perf_counter()
        # XXX or perhaps conjuction should be outside the timed section?
        conjunction = parse_one("(A AND B) OR (C AND D) OR (E AND F) OR (G AND H)")
        normalize.normalize(conjunction)
        elapsed += pyperf.perf_counter() - t0
    return elapsed
```

...and adjust the calls in `if __name__ == "__main__":` accordingly.

**Review by ericsnowcurrently — CHANGES_REQUESTED** [review]
Overall this looks good.  I do have some suggested changes.

**Comment by tobymao** [comment]
@ericsnowcurrently ready for another look.

and sure, i can add these benchmarks to the own suite

**Comment by tobymao** [comment]
> Relative to this benchmark specifically:
> 
> * it feels like an in-between one (not quite a macro-benchmark but more complex than a micro-benchmark)
> * could it be be made represent a full Python workload more closely (or integrated into such a benchmark)?
> * what workloads would it represent or be a part of?
> * how much coverage of those workloads are already in the pyperformance suite?
> * how should this benchmark be categorized/tagged?
> 
> (I'm sure we'll merge it in regardless of the answers.)

in terms of workflows, it represents a good chunk in that people want to parse many sql queries (data engineering / analytics). the normalizer also represents mutation of queries which is another kind of macro workflow. there are some companies that use sqlglot to parse 10s of thousands of sql queries to extract out metadata.

sqlglot has a prototype engine which could represent more macro workflows, but it's not quite ready yet and not something i want to expose at this point.

**Review by ericsnowcurrently — APPROVED** [review]
LGTM

**Comment by ericsnowcurrently** [comment]
Thanks for the benchmark!

**Pull request merged**